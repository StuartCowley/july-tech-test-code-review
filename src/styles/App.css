.App {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.nasa-logo {
  height: auto;
  width: 50%;
}

@media screen and (min-width: 1024px) {
  .nasa-logo {
    width: 30%;
  }
}