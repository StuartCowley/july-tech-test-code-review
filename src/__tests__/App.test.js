import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  it("redners App component correctly", () => {
    const { asFragment } = render(<App/>);

    expect(asFragment()).toMatchSnapshot(); 
  });

  it("renders logo", () => {
    render(<App />);
    const imgElement = screen.getByAltText("nasaLogo");

    expect(imgElement).toBeInTheDocument();
    expect(imgElement).toHaveClass("nasa-logo");
  });
});
