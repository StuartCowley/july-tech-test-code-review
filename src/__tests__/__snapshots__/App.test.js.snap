// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App redners App component correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="nasaLogo"
      class="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <div
      class="Search"
    >
      <form
        class="search-form"
      >
        <input
          class="search-input"
          type="text"
        />
        <button
          class="search-btn"
          type="submit"
        >
          Go!
        </button>
      </form>
    </div>
    <p>
      No results
    </p>
  </div>
</DocumentFragment>
`;

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="nasaLogo"
      class="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <div
      class="Search"
    >
      <form
        class="search-form"
      >
        <input
          class="search-input"
          type="text"
        />
        <button
          class="search-btn"
          type="submit"
        >
          Go!
        </button>
      </form>
    </div>
    <p>
      Search Results
    </p>
    <img
      alt="space-image"
      class="card-image"
      src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
    />
  </div>
</DocumentFragment>
`;
